modules/subscriptions/views.php

```diff
@@ -0,0 +1,754 @@
+					$subs_count = array(
+						'status'  => 'failed',
+						'code'    => $xml->getErrorCode(),
+						'message' => $xml->getErrorMessage(),
+						'value'   => ( isset( $subs_count['value'] ) ) ? $subs_count['value'] : 0,
+					);
+				} else {
+					$subs_count = array(
+						'status' => 'success',
+						'value'  => $xml->getResponse(),
+					);
+				}
+
+				set_transient( 'wpcom_subscribers_total', $subs_count, 3600 ); // try to cache the result for at least 1 hour
+			}
+		}
+
+		/**
+		 * Filter the total amount of subscribers
+		 *
+		 * @module subscriptions
+		 *
+		 * @since 6.9
+		 *
+		 * @param int|array $subscribed Information about the total amount of subscribers.
+		 */
+		return apply_filters( 'jetpack_subscription_widget_total_subscribers', $subs_count );
+	}
+
+	/**
+	 * Updates a particular instance of a widget when someone saves it in wp-admin.
+	 *
+	 * @param array $new_instance
+	 * @param array $old_instance
+	 *
+	 * @return array
+	 */
+	function update( $new_instance, $old_instance ) {
+		$instance = $old_instance;
+
+		if ( self::is_jetpack() ) {
+			$instance['title']                 = wp_kses( stripslashes( $new_instance['title'] ), array() );
+			$instance['subscribe_placeholder'] = wp_kses( stripslashes( $new_instance['subscribe_placeholder'] ), array() );
+			$instance['subscribe_button']      = wp_kses( stripslashes( $new_instance['subscribe_button'] ), array() );
+			$instance['success_message']       = wp_kses( stripslashes( $new_instance['success_message'] ), array() );
+		}
+
+		if ( self::is_wpcom() ) {
+			$instance['title']               = strip_tags( stripslashes( $new_instance['title'] ) );
+			$instance['title_following']     = strip_tags( stripslashes( $new_instance['title_following'] ) );
+			$instance['subscribe_logged_in'] = wp_filter_post_kses( stripslashes( $new_instance['subscribe_logged_in'] ) );
+			$instance['subscribe_button']    = strip_tags( stripslashes( $new_instance['subscribe_button'] ) );
+		}
+
+		$instance['show_subscribers_total']     = isset( $new_instance['show_subscribers_total'] ) && $new_instance['show_subscribers_total'];
+		$instance['show_only_email_and_button'] = isset( $new_instance['show_only_email_and_button'] ) && $new_instance['show_only_email_and_button'];
+		$instance['subscribe_text']             = wp_filter_post_kses( stripslashes( $new_instance['subscribe_text'] ) );
+
+		return $instance;
+	}
+
+	/**
+	 * The default args for rendering a subscription form.
+	 *
+	 * @return array
+	 */
+	static function defaults() {
+		$defaults = array(
+			'show_subscribers_total'     => true,
+			'show_only_email_and_button' => false
+		);
+
+		if ( self::is_jetpack() ) {
+			$defaults['title']                 = esc_html__( 'Subscribe to Blog via Email', 'jetpack' );
+			$defaults['subscribe_text']        = esc_html__( 'Subscribe to Blog via Email', 'jetpack' );
```

Comment:
Those default values do not appear to be correct; they are all the same

Reply:
great catch!